docs/schemapack_spec.md

```diff
@@ -0,0 +1,168 @@
+# Schemapack Specification
+
+This document describes the structure and components of a `schemapack.yaml` file, which defines the schema used to validate and interpret datapacks. Each schemapack includes:
+
+* A schemapack version identifier
+
+* An optional schema description
+
+* A classes section, where each class represents a distinct entity type with associated content and its relations. Classes define the internal structure (content) of their instances using JSON Schema and may specify relationships (relations) to other classes defining constraints on how instances of different classes are linked, including their multiplicity and participation.
+
+### Keywords:
+
+`schemapack`: Specifies the Schemapack version the schema is built under.
+
+`description` *(optional)*: A short, human-readable summary of the schema.
+
+`classes`: A mapping of class names to *class definitions*. Class names must use PascalCase.
+
+Example:
+```yaml
+
+schemapack: 3.0.0
+description: Schema for experimental metadata
+classes:
+    Experiment:
+        ...
+```
+
+`rootClass` *(optional)*: Defines the name of a class that should acting as the root of the schemapack.
+
+<div style="border: 1px solid #007acc; border-left: 4px solid #007acc; padding: 1em; border-radius: 6px; background: transparent; margin-bottom: 1em;">
+  <strong>Note:</strong> A rooted <code>schemapack</code> requires its corresponding <code>datapack</code> to be rooted as well. If you're validating a rooted <code>datapack</code> agains a schema, make sure that its <code>schemapack</code> counterpart is also rooted. If not specified , i.e. set to None (the default), the datapack must not specify a root resource.
```

Comment:
Same weird formatting in the rich diff as in the datapack case, might need to check that locally